pinot-core/src/main/java/com/linkedin/pinot/core/common/BlockValSet.java

```diff
@@ -28,6 +28,28 @@
   DataType getValueType();
 
   /**
+   * Get values for single-valued column.
+   *
+   * @param <T> Return type
+   * @return Values for single-valued column.
+   */
+  <T> T getSingleValues();
+
+  /**
+   * Get values for multi-valued column.
+   *
+   * @param <T> Return type
+   * @return Values for multi-valuted column.
+   */
+  <T> T getMultiValues();
```

Comment:
Add a TODO on this API. We need to re-think whether we should have batch reading for multi-values.